src/idl_parser.cpp

```diff
@@ -3290,18 +3300,15 @@ CheckedError Parser::ParseRoot(const char *source, const char **include_paths,
   return NoError();
 }
 
-CheckedError Parser::CheckPrivateLeak(const Definition &def) {
+CheckedError Parser::CheckPrivateLeak(const Definition &def,
+                                      const Definition &value_type) {
   if (!opts.no_leak_private_annotations) return NoError();
-
   const auto is_private = def.attributes.Lookup("private");
-  if (module_annotation_ == kNone) {
-    module_annotation_ = is_private ? kPrivate : kPublic;
-  }
-  if ((module_annotation_ == kPublic && is_private) ||
-      (module_annotation_ == kPrivate && !is_private)) {
+  const auto is_field_private = def.attributes.Lookup("private");
+  if (!is_private && is_field_private) {
     return Error(
-        "Leaking private implementation, verify all objects have similar "
-        "annotations");
+      "Leaking private implementation, verify all objects have similar "
+      "annotations");
   }
   return NoError();
 }
```

Comment:
@CasperN So with this approach we actually check each field by itself and return directly if we find a leaked annotation. and the test cases can also reflect that